File: Python/SortListTest.py
from unittest import TestCase

from SortList import SortList, ListNode


class TestSortList(TestCase):
    def test_sortList(self):
        sl = SortList()

        self.assertIsNone(sl.sortList(None))

        self.assertEqual(ListNode(0), sl.sortList(ListNode(0)))

        node1 = ListNode(1)
        node1.next = ListNode(0)

        node2 = ListNode(0)
        node2.next = ListNode(1)

        self.assertEqual(node2, sl.sortList(node1))

        node3 = ListNode(0)
        node3.next = ListNode(1)

        self.assertEqual(node3, sl.sortList(node2))

File: Golang/SortList/SortList_test.go
package SortList

import (
	"testing"
	"reflect"
)

func TestSortList(t *testing.T) {

	if sortList(nil) == nil {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(
		&ListNode{
			Val: 0,
		},
		sortList(&ListNode{
			Val: 0,
		})) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	node1 := &ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 0,
		},
	}
	node2 := &ListNode{
		Val: 0,
		Next: &ListNode{
			Val: 1,
		},
	}
	if reflect.DeepEqual(node2, sortList(node1)) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	node3 := &ListNode{
		Val: 0,
		Next: &ListNode{
			Val: 1,
		},
	}
	if reflect.DeepEqual(node3, sortList(node2)) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}
